linera-chain/src/manager.rs

```diff
@@ -60,34 +72,35 @@ use crate::{
 };
 use linera_base::{
     crypto::{KeyPair, PublicKey},
-    data_types::{BlockHeight, RoundNumber, Timestamp},
-    ensure,
+    data_types::{ArithmeticError, BlockHeight, Round, Timestamp},
+    doc_scalar, ensure,
     identifiers::{ChainId, Owner},
 };
-use linera_execution::committee::Epoch;
+use linera_execution::{committee::Epoch, ChainOwnership};
 use rand_chacha::{rand_core::SeedableRng, ChaCha8Rng};
 use rand_distr::{Distribution, WeightedAliasIndex};
 use serde::{Deserialize, Serialize};
-use std::{
-    collections::{BTreeMap, HashMap},
-    time::Duration,
-};
+use std::time::Duration;
 use tracing::error;
 
 const TIMEOUT: Duration = Duration::from_secs(10);
 
-/// The consensus state of a chain with multiple owners some of which are potentially faulty.
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct MultiOwnerManager {
-    /// The public keys of the chain's co-owners, with their weights.
-    pub public_keys: BTreeMap<Owner, (PublicKey, u64)>,
-    /// The number of initial rounds in which all owners are allowed to propose blocks,
-    /// i.e. the first round with only a single leader.
-    pub multi_leader_rounds: RoundNumber,
+/// The result of verifying a (valid) query.
+#[derive(Eq, PartialEq)]
+pub enum Outcome {
+    Accept,
+    Skip,
+}
+
+/// The state of the certification process for a chain's next block.
```

Comment:
nit: (here and in other places) Would `BlockManager` be a better name than `ChainManager`?

Reply:
Or `ConsensusManager`?
Since it manages neither the whole chain, nor all the blocks. It only manages the decision process for the next block.